### ISSUE TITLE
Support single user mode during migration for SQL Server 

### ISSUE THREAD

**Issue by reitzmichnicht** [flyway/flyway#1456]
###### What version of Flyway are you using?

4.0.3
###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin, SBT plugin, ANT tasks)

Java-API
###### What database are you using (type & version)?

SQL Server 2012
###### What operating system are you using?

Windows 7 64bit
###### What did you do?

(Please include the content causing the issue, any relevant configuration settings, and the command you ran)
Use this command inside migration script
alter database 'foo' set single_user with rollback immediate
...
alter database 'foo' set multi_user
###### What did you expect to see?

successful migration
###### What did you see instead?

exception because flyway has 2 DB connections during miration, one for metadata and one for the migration. The metadata connection should be closed during migration

**Comment by axelfontaine** [comment]
I currently see no solution to this as this statement must be run outside of a transaction with no other transaction active in the system, which at the very least would be forfeiting the ability to safely migrate from multiple nodes in parallel.

Using a different kind of locking like application locks is not an option either as these must also be acquired within a transaction.

**Comment by reitzmichnicht** [comment]
Yes transactions are a problem, but it would be possible to commit the running transaction before and start another one after the statements in sql, so Flyway should not feel any difference.

**Comment by reitzmichnicht** [comment]
Please add no transaction detection for alter database statements in sql server like in #1027

**Comment by axelfontaine** [comment]
Duplicate of #1426
